crowbar_framework/app/controllers/application_controller.rb

```diff
@@ -33,6 +33,9 @@ class ApplicationController < ActionController::Base
     Crowbar::Installer.successful? || \
     Rails.env.test?
   }
+  before_filter :upgrade, if: proc {
+    File.exist?("/var/lib/crowbar/upgrade/progress.yml")
```

Comment:
So, the library could have *another* file indicating that the upgrade is currently running, like `/var/lib/crowbar/upgrade/running`. That would be deleted by progress library when the library reaches the last step (and it succeeds). 
It's obviously not adding any new information to what we already have, its only reason would be so the check here in the `application_controller.rb` could be as simple (and fast) as possible.